internal/pkg/runtime/engines/singularity/process_linux.go

```diff
@@ -321,6 +321,11 @@ func (engine *EngineOperations) StartProcess(masterConn net.Conn) error {
 						sylog.Debugf("No child process, exiting ...")
 						os.Exit(128 + int(signal))
 					}
+				} else if engine.EngineConfig.GetSignalPropagation() {
+					if err := syscall.Kill(cmd.Process.Pid, signal); err == syscall.ESRCH {
```

Comment:
technically, it's possible that there's an error other than syscall.ESRCH. There should be a test for a non-nil error.

Reply:
technically yes but they are unlikely to happen in this context, `EINVAL` means Go runtime received an invalid signal number. `EPERM` means PID 1 process can't send signal to its direct child which is not possible